### ISSUE TITLE
HTTP/1 connections over TLS leave CLOSE_WAIT sockets lying around

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8766]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8766|
|Type|defect|
|Created|2016-08-18 11:45:27Z|
|Branch|https://github.com/twisted/twisted/tree/8766-computers-are-hard|

<details><summary>Searchable metadata</summary>

```
trac-id__8766 8766
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8766_computers_are_hard 8766-computers-are-hard
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1471520727328359 1471520727328359
changetime__1473067011941827 1473067011941827
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hynek|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_3"></a>|@hynek set owner to @hawkowl|
|-|-|

code review happened in private, LGTM and good to go once the builders are green.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset e133becc4502708ebf3c1ab1cf5e17e50bc637ef
```
#!CommitTicketReference repository="" revision="e133becc4502708ebf3c1ab1cf5e17e50bc637ef"
Merge 8766-computers-are-hard: Fix HTTP/1 + TLS connections never closing

Author: hawkowl
Reviewer: hynek
Fixes: #8766
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

In changeset 0c7c0e11dc896372c47dafe91f1c57421bbdf1b8
```
#!CommitTicketReference repository="" revision="0c7c0e11dc896372c47dafe91f1c57421bbdf1b8"
Merge 8766-computers-are-hard: Fix HTTP/1 + TLS connections never closing

Author: hawkowl
Reviewer: hynek
Fixes: #8766
```

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_7"></a>|@hynek commented|
|-|-|

FYI this is [CVE-2016-7100](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-7100).
